src/InfluxDB.Collector/Util/PseudoHighResTimestampSource.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+
+namespace InfluxDB.Collector.Util
+{
+
+
+
+    /// <summary>
+    /// Implements <see cref="ITimestampSource"/>
+    /// in a way that combines the low-ish resolution DateTime.UtcNow
+    /// with a sequence number added to the ticks to provide 
+    /// pseudo-tick precision timestamps
+    /// </summary>
+    /// <remarks>
+    /// See https://github.com/influxdata/influxdb-csharp/issues/46 for why this is necessary.
+    /// Long story short: 
+    /// a) InfluxDB has a "LastWriteWins" policy for points that have the same timestamp and tags. 
+    /// b) The normal <see cref="System.DateTime.UtcNow"/> only supplies timestamps that change not as often as you think.
+    /// c) In a web server, it's entirely possible for more than one thread to get the same UtcNow value
+    /// 
+    /// As a remediation for this, we infuse DateTime.UtcNow with a sequence number until it ticks over.
+    /// 
+    /// </remarks>
+    public class PseudoHighResTimestampSource : ITimestampSource
```

Comment:
Since we don't currently provide an API for substituting this in, can we keep this internal for now and avoid bikeshedding on the name? :-)

Reply:
_bikeshedding_ is an awesome term for this. Thanks for bringing it into my vocabulary :-)